packages/aila/src/core/Aila.ts

```diff
@@ -53,9 +57,21 @@ export class Aila implements AilaServices {
       promptBuilder: options.promptBuilder,
     });
 
-    this._lesson = new AilaLesson({ lessonPlan: options.lessonPlan ?? {} });
     this._prisma = options.prisma ?? globalPrisma;
 
+    this._lesson = new AilaLesson({
+      aila: this,
```

Comment:
Have seen some fiddly bugs arise from passing `aila: this` as a prop, as the types aren't necessarily accurate (due to a quasi race condition on instantiation)

Reply:
Not saying don't do this, just highlighting because I think you were away when it was discovered

Reply:
Okay that's interesting! I see. So if you've not yet set some property on the instance before you passing it, then it could cause a problem.